methods_exploration/it_functions.py

```diff
@@ -99,3 +126,98 @@ def calcMI_crit(M, nbins = 11, alpha = 0.05, numiter = 1000, ncores = 2):
     MIcrit = MIss[round((1-alpha)*numiter)] 
     return(MIcrit)
 
+def lag_data(M, shift):
+    '''lags data by shift for transfer entropy calculation
+    M: a numpy array of shape (nobs, 2) where nobs is the number of observations
+    this assumes that the data are arrange such that the first column is the source and
+    the second column is the sink.
+    shift: the number of time steps you want to lag the sink by, must be a positive integer
```

Comment:
This is a different lag, which is one of the reasons that I was calling this parameter `shift` instead of `lag`. The `n_lags` in the config is for looking at lagged correlation and lagged mutual information, not so different I suppose